frontend/src/app/modules/main/user-profile/user-profile.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, OnInit } from '@angular/core';
+import { Song } from 'src/app/models/song/song';
+import { SongsService } from 'src/app/services/songs/songs.service';
+
+@Component({
+  selector: 'app-user-profile',
+  templateUrl: './user-profile.component.html',
+  styleUrls: ['./user-profile.component.sass']
+})
+export class UserProfileComponent implements OnInit {
+  public isProfileMenuShown: boolean = false;
+
+  private readonly _scrollingSize: number = 270;
+  public topArtists = new Array<object>();
+
+  songs: Song[] = [];
+
+  constructor(
+    private _songService: SongsService
+  ) {}
+
+  ngOnInit(): void {
+    this.topArtists = this.getTopArtists();
+    this.loadTopSongs();
+  }
+
+  getTopArtists = (): Array<object> => new Array<object>();
```

Comment:
Use interface instead of an object.

Reply:
Hi, i have a question: if i use an interface instead of an object _public topArtists : Artist[]_, then i need to do _getTopArtists()_ similarly to _loadTopSongs()_, but we have no artist service yet. What should i do?

Reply:
No, that's will be a separate task

Reply:
I've created artist service to mock top artists loading, because i'm not sure about endpoints and i don't know how to mock it correctly in other way. It can be rewritten soon. Is it okay?